File: common_sdk/src/main/java/com/app/base/widget/dialog/LoadingDialog.java
package com.app.base.widget.dialog;

import android.content.Context;
import android.text.TextUtils;
import android.widget.TextView;

import androidx.annotation.StringRes;
import androidx.appcompat.app.AppCompatDialog;

import com.app.base.R;


/**
 * @author Ztiany
 * Date : 2017-05-02 14:37
 */
public class LoadingDialog extends AppCompatDialog {

    private TextView mMessageTv;

    public LoadingDialog(Context context) {
        super(context);
        setView();
    }

    private void setView() {
        setContentView(R.layout.dialog_loading);
        mMessageTv = findViewById(R.id.dialog_loading_tv_title);
    }

    public void setMessage(CharSequence message) {
        if (!TextUtils.isEmpty(message)) {
            mMessageTv.setText(message);
        }
    }

    public void setMessage(@StringRes int messageId) {
        if (messageId != 0) {
            mMessageTv.setText(messageId);
        }
    }

}

File: common/core/src/main/java/com/app/base/widget/dialog/loading/LoadingDialog.kt
package com.app.base.widget.dialog.loading

import android.app.Dialog
import android.content.Context
import android.text.TextUtils
import android.view.LayoutInflater
import androidx.annotation.StringRes
import androidx.appcompat.app.AppCompatDialog
import com.app.base.databinding.DialogLoadingBinding
import com.app.base.widget.dialog.common.showCompat

/**
 * @author Ztiany
 */
internal class LoadingDialog(context: Context) : AppCompatDialog(context), LoadingDialogInterface {

    private val vb = DialogLoadingBinding.inflate(LayoutInflater.from(context))

    init {
        setView()
    }

    private fun setView() {
        setContentView(vb.root)
    }

    fun setMessage(message: CharSequence?) {
        if (!TextUtils.isEmpty(message)) {
            vb.dialogLoadingTvTitle.text = message
        }
    }

    fun setMessage(@StringRes messageId: Int) {
        if (messageId != 0) {
            vb.dialogLoadingTvTitle.setText(messageId)
        }
    }

    override fun show() {
        this.showCompat {
            super@LoadingDialog.show()
        }
    }

    override val dialog: Dialog
        get() = this

    override fun updateMessage(message: CharSequence) {
        setMessage(message)
    }

}